### DISCUSSION TITLE
Global Dependencies doc example not working as-is

### DISCUSSION THREAD

**Discussion by jerinpetergeorge** [fastapi/fastapi#6411] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# code snippet copied from the official documentation

from fastapi import Depends, FastAPI, Header, HTTPException


async def verify_token(x_token: str = Header()):
    if x_token != "fake-super-secret-token":
        raise HTTPException(status_code=400, detail="X-Token header invalid")


async def verify_key(x_key: str = Header()):
    if x_key != "fake-super-secret-key":
        raise HTTPException(status_code=400, detail="X-Key header invalid")
    return x_key


app = FastAPI(dependencies=[Depends(verify_token), Depends(verify_key)])


@app.get("/items/")
async def read_items():
    return [{"item": "Portal Gun"}, {"item": "Plumbus"}]


@app.get("/users/")
async def read_users():
    return [{"username": "Rick"}, {"username": "Morty"}]
```

### Description

**TL;DR**: maybe related to https://github.com/tiangolo/fastapi/pull/4906

I was following the documentation of [**Global Dependencies**](https://fastapi.tiangolo.com/tutorial/dependencies/global-dependencies/#global-dependencies) and the example in the documentation is not working as expected.

I am getting the `422` status code with the following response
```json
{
    "detail": [
        {
            "loc": [
                "header",
                "x-token"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        },
        {
            "loc": [
                "header",
                "x-key"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```

Adding a ***`default`*** parameter to the `Header(...)` solved the issue

```python
async def verify_token(x_token: str = Header(default=None)):
    ...


async def verify_key(x_key: str = Header(default=None)):
    ...
```

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.8.10

**Comment by odiseo0** [comment, marked as answer]
If you're not passing the `X-Token` or the `X-Key` header you'll get a 422 error because you're telling the client that those headers are required
